bots/make-checkout

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python2
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import os
+import subprocess
+import sys
+
+sys.dont_write_bytecode = True
+
+# Check out the given ref and if necessary overlay the bots
+# directory on top of it as expected on non-master branches
+
+BOTS = os.path.dirname(__file__)
+BASE = os.path.normpath(os.path.join(BOTS, ".."))
+
+def main():
+    parser = argparse.ArgumentParser(description="Ensure necessary issue exists for translations")
+    parser.add_argument("-v", "--verbose", action="store_true", help="Verbose output")
+    parser.add_argument("--base", nargs='?', help="Base branch that revision is for")
+    parser.add_argument("ref", help="The git ref to fetch")
+    parser.add_argument("revision", default="FETCH_HEAD", nargs='?',
+            help="Actual commit to check out, defaults to ref")
+
+    opts = parser.parse_args()
+
+    def execute(*args):
+        if opts.verbose:
+            sys.stderr.write("+ " + " ".join(args) + "\n")
+        try:
+            output = subprocess.check_output(args, cwd=BASE)
+        except subprocess.CalledProcessError as ex:
+            sys.exit(ex.returncode)
```

Comment:
`ex` should be written to stderr here, too.

Reply:
The ex.message here is really useless: "subprocess exited with error code N". Note that CalledProcessError is not raised in scenarios where the subprocess can't be started. Only when it exits with an error.

Reply:
Right, my bad. If the executable doesn't exist, it's an `OSError`.